internal/consensus/reactor.go

```diff
@@ -48,8 +48,8 @@ type Reactor struct {
 	eventBus *types.EventBus
 
 	rsMtx         cmtsync.RWMutex
-	rs            *cstypes.RoundState
-	initialHeight int64 // under rsMtx
+	rs            cstypes.RoundState
+	initialHeight atomic.Int64
```

Comment:
Not regarding this PR, but the reason for which we need to keep this is so.... annoying and minor. I really would find a different solution for that.

Reply:
https://github.com/cometbft/cometbft/issues/4491